### PR TITLE
app-shells/starship: add 1.10.2

### PR DESCRIPTION
**Pull request by ixti**
* Starship replaced libgit2 based git2 crate with [git-repository](https://github.com/gentoo/gentoo/pull/25931), thus I've removed DEPEND (and RDEPEND) and relaxed down rust version requirement to `>= 1.59`
* Added MPL-2.0 to the LICENSE list (required by uluru - new dependency)

### CODE REVIEW

**Comment by ixti** [comment]
cc: @bowlofeggs

**Comment by ixti** [comment]
cc: @Flowdalic

**Comment by thesamesam** [comment]
Looks like gitoxide bundles zlib-ng. Not your fault, just unfortunate.

Also, could you rebase to 1.10.1?

**Comment by ixti** [comment]
> Looks like gitoxide bundles zlib-ng. Not your fault, just unfortunate.

:((

> Also, could you rebase to 1.10.1?

Sure!

**Comment by ixti** [comment]
@thesamesam done

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
This seems to work. The only change I think we need to make is the `~amd64` keyword.

I noticed that the git interactions seem slower than they were before, but I suppose that's an upstream concern ☺

File: app-shells/starship/starship-1.10.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+HOMEPAGE="https://starship.rs/"
+SRC_URI="
+	https://github.com/starship/starship/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+	$(cargo_crate_uris)
+"
+
+LICENSE="Apache-2.0 BSD BSD-2 CC0-1.0 ISC MIT MPL-2.0 Unlicense WTFPL-2 ZLIB"
+
+SLOT="0"
+KEYWORDS="amd64"
```
[the remark this answers is not in the document]
Comment (ixti) [reply app-shells/starship/starship-1.10.1.ebuild:392]: Oh. Thank you for catching this. Totally forgot about that.

File: app-shells/starship/starship-1.10.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,417 @@
+# Copyright 2022 Gentoo Authors
```
[the remark this answers is not in the document]
Comment (ixti) [reply app-shells/starship/starship-1.10.1.ebuild:1]: Good catch. Thank you.

File: app-shells/starship/starship-1.10.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+	https://github.com/starship/starship/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+	$(cargo_crate_uris)
+"
+
+LICENSE="Apache-2.0 BSD BSD-2 CC0-1.0 ISC MIT MPL-2.0 Unlicense WTFPL-2 ZLIB"
+
+SLOT="0"
+KEYWORDS="amd64"
+
+BDEPEND=">=virtual/rust-1.59"
```
[the remark this answers is not in the document]
Comment (ixti) [reply app-shells/starship/starship-1.10.1.ebuild:394]: Yes. Previous rust min version bump was due to git2 crate, that was replaced with gitoxide: https://github.com/starship/starship/pull/3883

File: app-shells/starship/starship-1.10.1.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (ixti) [reply app-shells/starship/starship-1.10.1.ebuild]: Fixed.

File: app-shells/starship/starship-1.10.1.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (ixti) [reply app-shells/starship/starship-1.10.1.ebuild]: Fixed.

**Comment by ixti** [comment]
> I noticed that the git interactions seem slower than they were before, but I suppose that's an upstream concern

Funny enough, [gitoxide](https://github.com/Byron/gitoxide)'s promise is: _perform as good or better than the canonical implementation_. :shrug:

**Comment by bowlofeggs** [comment]
I see that there is now also a 1.10.2 that upgrades gitoxide. There may be some performance benefits:

https://github.com/starship/starship/issues/4277

I think it's fine to merge this PR as is, and I also think it's fine to bump it up to 1.10.2. Up to you.

**Comment by ixti** [comment]
Oh. I think it's better to bump then. Will do that tomorrow.

**Comment by ixti** [comment]
@bowlofeggs Updated.

**Review by bowlofeggs — APPROVED** [review]

**Comment by bowlofeggs** [comment]
I don't have commit privileges, so we'll need someone to merge this for us.

**Comment by bowlofeggs** [comment]
https://bugs.gentoo.org/872584

**Pull request closed without merging**